MetaMask/metamask-extension
# fix: rename anonymous transaction events.

## Explanation
Currently anonymous and non-anonymous events share the same name. This creates confusion on data analysis done by stakeholders and blocks some of the automations and improvements that are being done by the data team. In order to improve this, we should rename the anonymous transactions events in MM to a different name so they can be easily identified and treated differently from the non-anonymous ones. 

### Acceptance Criterias
- [x] anonymous `Transaction Added` event is renamed to `Transaction Added Anon`
- [x] anonymous `Transaction Approved` event is renamed to `Transaction Approved Anon`
- [x] anonymous `Transaction Rejected` event is renamed to `Transaction Rejected Anon`
- [x] anonymous `Transaction Submitted` event is renamed to `Transaction Submitted Anon`
- [x] anonymous `Transaction Finalized` event is renamed to `Transaction Finalized Anon`
- [x] No change is made to non-anonymous transactions events
- [x] `Transaction Added Anon`, `Transaction Approved Anon`, `Transaction Rejected Anon`, `Transaction Submitted Anon`, `Transaction Finalized Anon` are added to the [transactions events documentation](https://github.com/ConsenSys/segment-schema/tree/main/libraries/events/metamask-transactions) in segment-schema

* Fixes [Issue](https://github.com/MetaMask/MetaMask-planning/issues/598)

## Manual Testing Steps
- launch Metamask
- open test-dapp
- do a send transaction
- cancel it
- do another send transaction
- confirm it
- check mix panel and see that Anonymous Transaction events now have Anon appended to the event names

## Pre-merge author checklist


## Pre-merge reviewer checklist
